packages/pa_ppx/pa_ppx.0.16/opam

```diff
@@ -60,6 +60,8 @@ depends: [
 conflicts: [
   "ocaml-option-bytecode-only"
   "pa_ppx_migrate" { <= "0.11" }
```

Comment:
Sorry for not noticing this in earlier prs. Why are these added as conflicts instead of having the correct upper bounds in their respective opam files?

Reply:
For the bytecode-only option, a reviewer suggested it b/c (IIRC) there was no way to indicate to only build bytecode binaries on a platform that lacked native-code compiler.

For the pa_ppx_migrate  constraint, it's outdated, and probably can be removed (I'll do so and see if the CI build goes thru -- I'm sorry about not noticing).

Reply:
OK, removing that pa_ppx_migrate constraint wasn't a problem.  can you let me know if I should move the other two to their respective packages ?

Reply:
[I deleted two replies, b/c they were invalidated by the results of CI runs.  I think this is an accurate answer]

For the bytecode-only option, a reviewer suggested it b/c (IIRC) there was no way to indicate to only build bytecode binaries on a platform that lacked native-code compiler.

For the others (e.g. for pa_ppx_migrate), I added them here b/c it seemed convenient.  But when I moved the constraint `pa_ppx < "0.16"` to the file pa_ppx_migrate.0.11, the CI build for that package failed on Debian, b/c the package `libpcre-dev` is no longer available on Debian.  The CI build failed when trying to build "conf-libpcre".

More modern versions of camlp5 and pa_ppx_* all rely on pcre2, not pcre, so that problem should be going away.

But this does bring up a mystery and maybe a problem: I have one package, `pa_ppx_regexp` that depends on all of `re`, `pcre`, and `pcre2`.  B/c it allows the user to choose which "backend" to use for their regexps.  I just released a new version of this package, which went thru fine (you approved it yesterday).  So I don't know why that went thru.

Concretely, and this is an answer I would not have had before going thru this (ahem) "process of discovery", the reason maybe it's useful to put these constraints here in the new package, is that it means you don't have to build the old packages that you might have modified otherwise, and that can mean avoiding doing a bunch of spurious builds that might even fail for entirely unrelated reasons.

Whew!

Sorry, hope this is a good-enough answer.  I'm writing this before the CI run completes, so hopefully I won't be deleting it and writing -another- reply.  *grin*

Reply:
OK, the CI run passed, as it did before.

Reply:
For the bytecode-only I agree, my question was concerning the packages. If they are incompatible with pa_ppx, I would drop conflicts here and move the upper bounds on the respective packages. It is easier for the solver to deal with them than via conflicts

Reply:
We can do that in 2 stages. First add the relevant upper bounds on the other packages, then remove the conflicts from this. I can help in a couple of days if you want. And sorry again for not noticing it before

Reply:
OK, I can move them, but then some of those respective packages will not build (or their deps/revdeps won't build -- is that OK?)

Reply:
That's all right. 

By the way, if you set an `x-maintenance-intent` field on your latest release, or if you flag the old ones as deprecated while you decide what you prefer to maintain, we can move them to the opam-repository-archive in the next stages of cleanup. See [Plan](https://discuss.ocaml.org/t/proposed-package-archiving-policy-for-the-opam-repository/15713) and recent [Announcement](https://discuss.ocaml.org/t/opam-repository-archival-phase-1-unavailable-packages/15797) for more details